app/models/concerns/foreman/thread_session.rb

```diff
@@ -37,6 +37,24 @@ def clear_thread
       end
     end
 
+    # This allows getting and setting all current values in case it's needed,
```

Comment:
I think this refactoring, while possibly good in the long run, is outside of the scope of this issue, which is needed for 1.15.3 (currently CSV export provides incorrect/partial information). 
I wouldn't want to modify the existing methods here in a minor release, as they are very critical in many places, therefor I only added functionality.